### ISSUE TITLE
Trailing comma in selector causes error with no-descending-specificity and Less syntax

### ISSUE THREAD

**Issue by eddiemoore** [stylelint/stylelint#1156]
Found a bug when running stylelint against a bunch of less files.

Error with: `no-descending-specificity`

Hard to say really what part of the code is causing this issue, as i'm running it agains a HUGE amount of files.

**Stylelint configuration:**
```npm install stylelint-config-migme```

```json
{
  "extends": "stylelint-config-migme"
}
```

Running with version `6.2.2` of stylelint

Running it with CLI with `stylelint app/styles/* --syntax less`

I believe it may relate to non-standard styntax because i'm running it against .less files

Expected warnings to be flagged and not causing it to crash.

The following warnings were flagged:
```shell
TypeError: Cannot read property 'filter' of undefined
    at lastCompoundSelectorWithoutPseudo (/XXX/node_modules/stylelint/dist/rules/no-descending-specificity/index.js:93:53)
    at checkSelector (/XXX/node_modules/stylelint/dist/rules/no-descending-specificity/index.js:34:39)
    at Processor.func (/XXX/node_modules/stylelint/dist/rules/no-descending-specificity/index.js:26:20)
    at Processor.process (/XXX/node_modules/postcss-selector-parser/dist/processor.js:31:14)
    at /XXX/node_modules/stylelint/dist/rules/no-descending-specificity/index.js:27:14
    at Array.forEach (native)
    at /XXX/node_modules/stylelint/dist/rules/no-descending-specificity/index.js:24:69
    at Array.forEach (native)
    at /XXX/node_modules/stylelint/dist/rules/no-descending-specificity/index.js:20:22
    at /XXX/node_modules/postcss/lib/container.js:110:28
```

**Comment by evilebottnawi** [comment]
@eddiemoore show example of code and next format issue as template

**Comment by davidtheclark** [comment]
The stack trace is a good indication of what's going wrong but we don't know what sourcecode caused it. The function that's throwing is this:

```js
function lastCompoundSelectorWithoutPseudo(selectorNode) {
  const nodesAfterLastCombinator = _.last(selectorNode.nodes[0].split(node => {
    return node.type === "combinator"
  }))

  const nodesWithoutPseudos = nodesAfterLastCombinator.filter(node => {
    return node.type !== "pseudo"
  })

  return nodesWithoutPseudos.toString()
}
```

So somehow we're ending up with `nodesAfterLastCombinator` being undefined or something. We could build safeguards into that function blindly if we're unable to pinpoint the Less that causes it.

**Comment by eddiemoore** [comment]
When I get into work tomorrow morning I'll see if I can narrow down the files a bit to see if I can pinpoint the piece of LESS code that's causing the problem

**Comment by eddiemoore** [comment]
Ok. I have narrowed it down to a single section.

```css
.selector,
{
  color: red;
}
```

Seems that if there is a comma at the end of the selectors list then it runs into the error. However LESS still compiles the code with no problems. Would have thought the less compiler should point out this issue.

**Comment by evilebottnawi** [comment]
@eddiemoore it is not valid CSS, @davidtheclark what do you think we should consider it as an additional check?

**Comment by eddiemoore** [comment]
@evilebottnawi Yeah I know it's not valid CSS. It's in legacy code, found it strange that the less compiler even compiled it.

**Comment by evilebottnawi** [comment]
@eddiemoore waiting what say @davidtheclark about this issue

**Comment by davidtheclark** [comment]
I think we can try to refactor that function just a little so it is more elegantly handled. Shouldn't be a big deal.

**Comment by evilebottnawi** [comment]
Close. Merged. Ref: https://github.com/stylelint/stylelint/pull/1181
